inter-nerdfont: init at 4.0

midischwarz12 opened the pull request:
## Description of changes

Adding NerdFont patch of Inter to nixpkgs.

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#reviewing-contributions
-->

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

getchoo committed:
key-rack: init at 0.4.0

thiagokokada committed:
linuxKernel.kernels.linux_zen: sync config with upstream

999eagle committed:
python312Packages.capstone_4: enable, add distutils patch

fabaff committed:
Merge pull request #326197 from fabaff/extract-msg-bump

python312Packages.extract-msg: 0.48.5 -> 0.48.7

K900 committed:
linux_testing: 6.10-rc6 -> 6.10-rc7

K900 committed:
linux_6_9: 6.9.8 -> 6.9.9

K900 committed:
linux_6_6: 6.6.37 -> 6.6.39

K900 committed:
linux_6_1: 6.1.97 -> 6.1.98

K900 committed:
linux-rt_5_10: 5.10.219-rt111 -> 5.10.220-rt112

K900 committed:
linux_latest-libre: 19584 -> 19597

K900 committed:
linux/hardened: fix syntax in update-script

jchv committed:
binwalk: migrate to pep517 builder

jchv committed:
binwalk: switch to fork, 2.3.4 -> 2.4.1

vlaci committed:
lief: 0.13.2 -> 0.14.1

Munksgaard committed:
livebook: 0.12.1 -> 0.13.3

Aleksanaa committed:
Merge pull request #315041 from kilianar/stellarium-version-test

stellarium: add passthru.tests.version

johnjameswhitman committed:
ib-{controller,tws}: mark as broken (#324591)

* Marks ib-tws broken.

* Marks ib-controller broken.

* Relocates release note to derivation comment.

---------

Co-authored-by: John Whitman <john@s2sq.com>
Co-authored-by: aleksana <me@aleksana.moe>

Scriptkiddi committed:
oxidized: 0.29.1 -> 0.30.1

SuperSandro2000 committed:
Merge pull request #315892 from Moraxyc/artalk

natsukium committed:
markdownlint-cli2: 0.9.0 -> 0.13.0 (#317966)

Co-authored-by: Sandro <sandro.jaeckel@gmail.com>

K900 committed:
Merge pull request #326290 from K900/kernels-20240711

Kernel updates for 2024-07-11

Sigmanificient commented on pkgs/by-name/in/inter-nerdfont/package.nix:
```diff
@@ -0,0 +1,39 @@
+{ lib, pkgs, stdenvNoCC, fetchzip }:
+
+let
+  version = "4.0";
+in stdenvNoCC.mkDerivation {
+  pname = "inter-nerdfont";
+  inherit version;
+
+  src = fetchzip {
+    url = "https://github.com/rsms/inter/releases/download/v${version}/Inter-${version}.zip";
```
Use `finalAttrs` to allow versison overrides
```suggestion
stdenvNoCC.mkDerivation (finalAttrs: {
  pname = "inter-nerdfont";
  version = "4.0";

  src = fetchzip {
    url = "https://github.com/rsms/inter/releases/download/v${finalAttrs.version}/Inter-${finalAttrs.version}.zip";
```

(with extra parenthesis at the end)

Sigmanificient commented on pkgs/by-name/in/inter-nerdfont/package.nix:
```diff
@@ -0,0 +1,39 @@
+    runHook postInstall
+  '';
+
+  buildInputs = builtins.attrValues {
+    inherit (pkgs)
+      fontforge
+      nerd-font-patcher;
+  };
```
Can you import dependencies directly as arguments?
```suggestion
  buildInputs = [ fontforge nerd-font-patcher ];
```

cvoges12 replied:
@Sigmanificient Why is the `pkgs.` prefix un-needed?

eclairevoyant replied:
you'd have to change the package expression's argset as well

eclairevoyant commented on pkgs/by-name/in/inter-nerdfont/package.nix:
```diff
@@ -0,0 +1,37 @@
+{ lib, pkgs, stdenvNoCC, fetchzip }:
```
```suggestion
{ lib, fontforge, nerd-font-patcher, stdenvNoCC, fetchzip }:
```

eclairevoyant commented on pkgs/by-name/in/inter-nerdfont/package.nix:
```diff
@@ -0,0 +1,37 @@
+    hash = "sha256-hFK7xFJt69n+98+juWgMvt+zeB9nDkc8nsR8vohrFIc=";
+  };
+
+  installPhase = ''
+    runHook preInstall
+
+    mkdir -p $out/share/fonts/truetype
+    nerd-font-patcher Inter.ttc
+    cp 'Inter Nerd Font.ttc' $out/share/fonts/truetype/InterNerdFont.tcc
```
```suggestion
  buildPhase = ''
    runHook preBuild

    nerd-font-patcher Inter.ttc
    
    runHook postBuild
'';

  installPhase = ''
    runHook preInstall

    install -Dm444 'Inter Nerd Font.ttc' $out/share/fonts/truetype/InterNerdFont.ttc
```

eclairevoyant replied:
probably could also use `dontConfigure = true;`

eclairevoyant commented on pkgs/top-level/all-packages.nix:
```diff
@@ -28931,6 +28931,8 @@ with pkgs;
 
   inter = callPackage ../data/fonts/inter { };
 
+  inter-nerdfont = callPackage ../by-name/in/inter-nerdfont/package.nix { };
+
```
```suggestion
```

this is unneeded with by-name

eclairevoyant commented on pkgs/by-name/in/inter-nerdfont/package.nix:
```diff
@@ -0,0 +1,37 @@
+
+  meta = {
+    homepage = "https://gitlab.com/mid_os/inter-nerdfont";
+    description = "A NerdFont patch of the Inter font";
```
```suggestion
    description = "NerdFont patch of the Inter font";
```

eclairevoyant reviewed (dismissed):
Thanks for the contribution! I have some additional feedback below.

Also, please rebase + squash into two commits:
* `maintainers: add cvoges12`
* `inter-nerdfont: init at 4.0`

nicoonoclaste committed:
Merge #324867: mypy 1.10.0 → 1.10.1

dotlambda committed:
Merge pull request #326097 from dotlambda/python3Packages.imap-tools

python312Packages.imap-tools: 1.6.0 -> 1.7.0

K900 committed:
xwayland: 24.1.0 -> 24.1.1

scotttrinh committed:
gleam: 1.3.0 -> 1.3.2

Release: https://github.com/gleam-lang/gleam/releases/tag/v1.3.2
Changelog: https://github.com/gleam-lang/gleam/blob/v1.3.2/CHANGELOG.md
Diff: https://github.com/gleam-lang/gleam/compare/v1.3.0...v1.3.2

Lassulus committed:
Merge pull request #326313 from helsinki-systems/oxidized_0.30.1

oxidized: 0.29.1 -> 0.30.1

cvoges12 commented:
> Thanks for the contribution! I have some additional feedback below.
> 
> Also, please rebase + squash into two commits:
> 
> * `maintainers: add cvoges12`
> * `inter-nerdfont: init at 4.0`

Wouldn't I need to send in a new PR for this? @eclairevoyant

eclairevoyant commented:
No, you need to use `git rebase -i`

K900 committed:
Merge pull request #326340 from K900/xwayland-24.1.1

xwayland: 24.1.0 -> 24.1.1

thiagokokada committed:
Merge pull request #326216 from thiagokokada/linux-zen-sync-config

linuxKernel.kernels.linux_zen: sync config with upstream

dotlambda committed:
python312Packages.eq3btsmart: init at 1.1.9

dotlambda committed:
python312Packages.construct-typing: init at 0.6.2

dotlambda committed:
home-assistant: support eq3btsmart component

Artturin committed:
Merge pull request #240602 from uninsane/gthumb-webservices

gthumb: make the webservices feature be optional

mweinelt committed:
Merge pull request #326363 from dotlambda/python3Packages.eq3btsmart

home-assistant: support eq3btsmart component

jopejoe1 committed:
discord: updates (#326186)

* discord: 0.0.58 -> 0.0.59

* discord-ptb: 0.0.92 -> 0.0.93

* discord-canary: 0.0.438 -> 0.0.450

* discord-development: 0.0.21 -> 0.0.22

* pkgsCross.aarch64-darwin.discord: 0.0.309 -> 0.0.310

* pkgsCross.aarch64-darwin.discord-ptb: 0.0.121 -> 0.0.122

* pkgsCross.aarch64-darwin.discord-canary: 0.0.547 -> 0.0.559

* pkgsCross.aarch64-darwin.discord-development: 0.0.43 -> 0.0.44

vcunat committed:
Revert "mypy: 1.10.0 → 1.10.1"

samuela committed:
Merge pull request #325706 from r-ryantm/auto-update/python312Packages.google-cloud-artifact-registry

python312Packages.google-cloud-artifact-registry: 1.11.3 -> 1.11.4

vcunat committed:
Merge #326371: Revert "mypy: 1.10.0 → 1.10.1"

Unexplained large rebuild. Moving to staging.

superherointj committed:
Merge pull request #326341 from scotttrinh/gleam-1.3.2

gleam: 1.3.0 -> 1.3.2

Lassulus committed:
Merge pull request #326322 from vlaci/lief-0141

lief: 0.13.2 -> 0.14.1

risicle committed:
Merge pull request #326281 from 999eagle/fix/capstone_4

python312Packages.capstone_4: enable, add distutils patch

happysalada committed:
Merge pull request #326309 from Munksgaard/bump-livebook

livebook: 0.12.1 -> 0.13.3

superherointj committed:
Merge pull request #326295 from jchw-forks/fix/binwalk

binwalk: migrate to fork, 2.4.1

chiroptical committed:
erlang_27: 27.0 -> 27.0.1

chiroptical committed:
erlang: 25.3.2.12 -> 25.3.2.13

midischwarz12 committed:
maintainers: added midirhee12

NickCao committed:
Merge pull request #325732 from r-ryantm/auto-update/python312Packages.google-cloud-asset

python312Packages.google-cloud-asset: 3.26.1 -> 3.26.2

jeremyschlatter committed:
go-ethereum: set `meta.mainProgram = "geth"`

This lets `nix run nixpkgs#go-ethereum` do something useful, namely run
geth.

jeremyschlatter committed:
go-ethereum: migrate to pkgs/by-name

midischwarz12 committed:
inter-nerdfont: init at 4.0

corngood committed:
garnet: used combined dotnet-sdk

This allows the sdk dependencies to be removed from deps.nix, so the
package will build on newer sdks.

happysalada committed:
Merge pull request #322463 from chiroptical/erlang-26-2-5-1

erlang_26: 26.2.5 -> 26.2.5.1

happysalada committed:
Merge pull request #326403 from chiroptical/chiroptical/erlang-27-0-1

erlang: 25.3.2.12 -> 25.3.2.13; erlang_27: 27.0 -> 27.0.1

corngood committed:
Merge pull request #326408 from corngood/garnet

garnet: used combined dotnet-sdk

Aleksanaa committed:
Merge pull request #325381 from jeremyschlatter/go-ethereum_mainProgram_geth

go-ethereum: set `meta.mainProgram = "geth"`

cvoges12 commented:
@eclairevoyant I had some trouble because of needing to split commits. But I managed to rebase in the end. It is ready to merge now.

Aleksanaa committed:
Merge pull request #321743 from getchoo/pkgs/key-rack/init

key-rack: init at 0.4.0

cvoges12 commented:
@eclairevoyant Is there anything this is waiting on before merging?

cvoges12 commented:
@eclairevoyant @Sigmanificient Is there anything needed to get this merged?

midirhee12 commented:
Okay, my bad. This seemed to close itself after doing a force push. I'll have to recreate this PR.

midirhee12 commented:
Reopening.

eclairevoyant commented:
Locking due to mass ping; please feel free to open a new PR.
